src/mongo/db/storage/wiredtiger/wiredtiger_record_store_oplog_stones.h

```diff
@@ -129,12 +136,8 @@ class WiredTigerRecordStore::OplogStones {
     // database, and false otherwise.
     bool _isDead = false;
 
-    // Maximum number of stones to keep in the deque before the background reclaim thread should
-    // truncate the oldest ones. Does not include the stone currently being filled. This value
-    // should not be changed after initialization.
-    size_t _numStonesToKeep;
     // Minimum number of bytes the stone being filled should contain before it gets added to the
-    // deque of oplog stones. This value should not be changed after initialization.
```

Comment:
because in the original implementation, each oplog should have the same size which is calculated in the initialization of OplogStones, it is easy to check if oplogs exceeded the limit size by comparing _numStonesToKeep and currentOplogStones.size(). However, if oplogSize is changed, the new Stones should have a different size with the remaining ones, re-arranging remaining oplogStones costs too much so it can't be a possible way. So we have to iterate and sum size of each oplogStone, including the remaining ones and the newer ones, to check if oplog size exceeded.That's why I remove _numStonesToKeep and rewrite hasExcessStones into hasExcessStones_inlock